keanu-project/src/main/java/io/improbable/keanu/vertices/VertexId.java

```diff
@@ -0,0 +1,63 @@
+package io.improbable.keanu.vertices;
+
+import java.util.LinkedList;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+
+import com.google.common.primitives.Ints;
+
+/**
+ * An object representing the ID value of a vertex.  IDs are assigned in such a way that a Lexicographic ordering of
+ * all nodes based on their ID value is a valid topological ordering of the graph made up by those vertices.
+ *
+ * Ids also encapsulate the notion of "Depth".  When we have graphs within graphs, the depth tells us at what level the
+ * graph exists - ie depth 1 is the outermost graph, depth 2 is a graph within a graph etc.
+ */
+public class VertexId implements Comparable<VertexId> {
+
+    public static final AtomicLong ID_GENERATOR = new AtomicLong(0L);
+
+    LinkedList<Long> idValues = new LinkedList<>();
+
+    public VertexId() {
```

Comment:
If I understand this correctly, each vertex needs to keep track of its parents' id values? Should we add a non-default constructor that takes `VertexId`?

Reply:
It doesn't keep track of all of its parents values - the ID just becomes a sequence of longs dependent upon its position in the graphception hierarchy (ie if Vertex 10 is stored in BayesVertex ID 5, then the ID would be [5, 10] - completely unrelated to the parent vertices.

The stuff to actually construct these properly will come in as part of the full graphception PR - this is just the code to move things out such that everything works as it currently does.